### DISCUSSION TITLE
Sites location is empty

### DISCUSSION THREAD

**Discussion by zipswich** [GetPublii/Publii#1385] [Help] [answered]
It is my first day with Publii, so please forgive my ignorance. I did do some search before asking here.
I did the following:

1. 3-dot menu at the top right corner > File location > Site location > selected folder E:\Publii
2. I created a post and took a look at its preview. It looks good.

I am looking at E:\Publii now and it is empty. I expected all posts would be stored here.
Could anyone point out where misunderstanding is?

**Comment by 3muchfun** [comment]
Please have a look at this: https://getpublii.com/docs/publii-global-settings.html
You will find a section "Files Location", where you can see the path where the site(s) are being stored.

**Reply by zipswich** [reply]
@3muchfun Thank you for lending a hand. I checked the Files Location section and it says:
"Sites Location - Define where the files for the websites you create with Publii, including themes, are stored. This can be set to, for example, your Dropbox folder to backup your site in the cloud, or any local directory."
My understanding is that this is where the files corresponding to the posts I create are stored. However, the folder is still empty.

**Reply by 3muchfun** [reply]
At the top right corner klick on the three dots (!) and chooses App Settings (2) <img width="253" alt="Bildschirmfoto 2024-02-20 um 21 45 07" src="https://github.com/GetPublii/Publii/assets/17292904/1afd3210-9757-4771-82d9-e12f73854336">

Scroll down to Files location
<img width="996" alt="Bildschirmfoto 2024-02-20 um 21 47 20" src="https://github.com/GetPublii/Publii/assets/17292904/b3593868-944a-419c-a0b0-e007a706fbc5">

If you do not know the default site location, try a search with the name of your site. You should find a folder with the name of the site. Look at the properties of the folder for the path.

**Reply by zipswich** [reply]
@3muchfun Thank you for the screenshots. That is exactly what I did.
![image](https://github.com/GetPublii/Publii/assets/6095671/da52f773-5e2a-4fe5-8d6b-401c57e6d655)
![image](https://github.com/GetPublii/Publii/assets/6095671/358f950a-21cf-4c4d-9f0c-4d25dcb400ea)

I found many folders under C:\Users\h6233\Documents\Publii\sites\blog

Blog is the site's name. h6233 is the user name.

**Reply by 3muchfun** [reply]
Ok, now its is up to @bobmitro

**Reply by example-user30** [reply]
What is E:\ drive?  

When you relocate the location for your site files there is an option to copy those files.  It is presented in the negative form presuming that anyone changing their Sites location will want to copy all their site files.

If you read it wrong and enabled the toggle then there will be NO files in the destination folder.
![shot_240222_201555](https://github.com/GetPublii/Publii/assets/156127535/d3ef44b7-249f-4d8d-b128-2c2040a7fe81)
 This option should be disabled to move all the files from your local users folder to the new location.

Try it again.

**Reply by zipswich** [reply]
@example-user30 I will.
My computer has two drives: C and E. E is just a 2TB hard drive that has most of my files. 
As I mentioned in my question, I created new posts after changing the location but the new location is still empty 2 days after I changed the location.

If I remember correctly, I was prompted to confirm moving files and I accepted it.

**Reply by zipswich** [reply]
Not sure if I did it correctly. I recorded the screen of every step:
1. Created a new directory E:\Temp\Publii, then selected it:
![image](https://github.com/GetPublii/Publii/assets/6095671/9ec1601d-a30a-48ab-b1a2-88c3b4ac0d62)
2. Clicked OK when prompted:
![image](https://github.com/GetPublii/Publii/assets/6095671/04f98d54-3067-48f4-a19f-52bdfa96fe35)
3. I use the File Explorer to check the new directory:
![image](https://github.com/GetPublii/Publii/assets/6095671/e96f78e0-2905-481a-beac-3ee0ea14d50c)

Not sure whether I have missed a step or misunderstood something.

**Reply by example-user30** [reply, marked as answer]
You did exactly what I do.  However I put my files on my Google Drive.  I like that since it stores everything in the cloud and I will not lose my sites and data if my computer drives go TU.

I am curious: - 
1. what version of Publii you are using.  That probably does not matter - just curious.
2. How is your E:\ drive formatted?  NTFS?  What is the name of your website (which should be the name of your folder also)?
3.  Can you create a folder on the root of your C:\ drive and move the files location there?  Does it work?
4. Do you ever run Publii as administrator?  (right click and select Run as administrator) - that should be a different user, but maybe moving your files to E:\ might work as administrator.

**Comment by zipswich** [comment]
1. Publii version:  0.44.4 (build 16557)
2. Drive E:
![image](https://github.com/GetPublii/Publii/assets/6095671/7dba0f48-6ca1-40a4-976c-a41fe2b218d9)
3. Created C:\Publii, but no files were moved there. I created a website Test and a post called Test and the files are in C:\Publii.
![image](https://github.com/GetPublii/Publii/assets/6095671/2dbb0055-4153-42c1-8bc4-629ca5428704)
![image](https://github.com/GetPublii/Publii/assets/6095671/a70115d6-c689-466b-b7f7-fa1af6a07572)
![image](https://github.com/GetPublii/Publii/assets/6095671/73603f7a-edb0-429d-a105-48a28196b945)
4. I have started Publii in the default mode (non-administrator) only once and have never exited it.

I changed the location again to E:\RandD\Publii in exactly the same way as I did before, and everything worked as expected: contents of C:\Publii have been moved to E:\RandD\Publii (C:\Publii is empty now).

![image](https://github.com/GetPublii/Publii/assets/6095671/f2f5bafd-e8ca-4aaa-a200-9eb865522284)

However, the original location C:\Users\h6233\Documents\Publii\sites\blog still has all the folders:
![image](https://github.com/GetPublii/Publii/assets/6095671/74b75359-0d7e-4a45-9e24-a21880849a0e)




4.

**Reply by example-user30** [reply]
It looks like by one way or another it is working.  
May I ask a silly question?  You said:
"I have started Publii in the default mode (non-administrator) only once and have never exited it."
I am hoping you are referring just to this last troubleshooting session.  You are exiting and restarting Publii when you are experiencing issues - yes?

**Reply by zipswich** [reply]
No, I have never exited Publii since I started it a week ago. I do not shut down my computer.

**Reply by 3muchfun** [reply]
Maybe you should restart Publii. It uses a SQLite database I the background, and some changes to the App take effect if the App is restarted, only.

**Reply by zipswich** [reply]
I have just restarted Publii but the original location C:\Users\h6233\Documents\Publii\sites\blog still has all the contents.   The name of the initial site is "blog"
![image](https://github.com/GetPublii/Publii/assets/6095671/dc88c111-f1bf-4773-b502-ee208e952556)

Again, the previous location (C:\publii) was emptied after I changed the location and its contents were moved to the new location on drive E.

This is the new location with the site called "test" on Drive E:
![image](https://github.com/GetPublii/Publii/assets/6095671/18d8d9ca-716a-407d-96bb-13f441517409)

Wonder if anyone could offer me a tip on how to handle the output in the simplest way. I maintain this site on one computer and the server is on the same LAN, so I can access the server as a network drive and copy files freely.  I know that I can use FTP for the server. Can I simply use the file system to output files to a directory?

**Reply by example-user30** [reply]
Take the output question to a new thread so they don't get confused too much.

I have to admit I am confused.  

If you close and open Publii are your files on E:\ now?

**Reply by zipswich** [reply]
Yes, the files are on drive E now as stated previously. 
I have set up FTP and been using it every day. 
I will start a new thread regarding direct file copying which is not a big deal.
